Desktop/AddGoalWindow.cs

```diff
@@ -51,78 +53,138 @@ private void PopulateGoalListView()
             PopulateGoalListView(GoalList);
         }
 
+        public string GoalPossibility(int profit, double worth)
+        {
+            double profitAWeek = profit / 4;
+            if (worth / profitAWeek <= 0.5)
+            {
+                return "Huge";
+            }
+            else if (worth / profitAWeek <= 0.8)
+            {
+                return "Real";
+            }
+            else if (worth / profitAWeek <= 1)
+            {
+                return "Small";
+            }
+            else
+            {
+                return "Not real";
+            }
+
+        }
+
         private void PopulateGoalListView(IEnumerable<Goal> GoalList)
         {
             listView1.Items.Clear();
 
             foreach (var goal in GoalList)
             {
+                int money;
+                if (((DateTime)goal.Deadlinedate).Subtract(DateTime.UtcNow).Days > 7)
+
+                    money = (Decimal.ToInt32(goal.Amount)) / ((((DateTime)goal.Deadlinedate).Subtract(DateTime.UtcNow) / 7).Days);
+
+                else
+                    money = (Decimal.ToInt32(goal.Amount));
+
+                int profit = 200;
+                string possibility = GoalPossibility(profit, money);
                 var item = new ListViewItem(new string[] {
                     goal.Title,
-                    ((DateTime) goal.GoalDate).ToString("yyyy-MM-dd"),
+                    ((DateTime) goal.Deadlinedate).ToString("yyyy-MM-dd"),
                     goal.Amount.ToString(),
                     goal.Description,
+                    money.ToString(),
+                    possibility
                 });
 
                 listView1.Items.Add(item);
             }
         }
+
         private void goalAmount_KeyPress(object sender, KeyPressEventArgs e)
         {
-            char ch = e.KeyChar;
+            var ch = e.KeyChar;
 
             if (ch == 46 && goalMoney.Text.IndexOf('.') != -1)
             {
                 e.Handled = true;
                 return;
             }
+
             if (!Char.IsDigit(ch) && ch != 8 && ch != 46)
             {
                 e.Handled = true;
             }
         }
 
-        public void ValidateFields(string amount, string details)
+        public void ValidateFields(string amount, string details, string name)
         {
             if (String.IsNullOrWhiteSpace(amount))
             {
-                goalMoney.BackColor = Color.Red;
+                  goalMoney.BackColor = Color.Red;
             }
+
             if (String.IsNullOrWhiteSpace(details))
             {
-                goalMoney.BackColor = Color.Red;
+                descriptionBox.BackColor = Color.Red;
+            }
+
+            if (String.IsNullOrWhiteSpace(name))
+            {
+                goalNameBox.BackColor = Color.Red;
             }
         }
+
         private void button1_Click(object sender, EventArgs e)
```

Comment:
Button should have a specific name.

Reply:
Thank you for noticing, i will rename it